addons/mrp/migrations/10.0.2.0/post-migration.py

```diff
@@ -0,0 +1,230 @@
+from openupgradelib import openupgrade
+
+
+def update_mrp_workcenter_times(cr):
+    """The time now in minutes instead of hours."""
+    cr.execute(
+        """
+        UPDATE mrp_workcenter
+        SET time_start = COALESCE(time_start, 0) * 60,
+            time_stop = COALESCE(time_stop, 0) * 60
+        """
+    )
+
+
+def map_mrp_production_state(cr):
+    """Get ids of old 'draft' MOs and map states.
+    :return: list of ids with all old draft MOs.
+    """
+    cr.execute(
+        """
+        SELECT id
+        FROM mrp_production
+        WHERE state = 'draft'
+        """
+    )
+    draft_ids = [r[0] for r in cr.fetchall()]
+    openupgrade.map_values(
+        cr,
+        openupgrade.get_legacy_name('state'), 'state',
+        [('draft', 'confirmed'),
+         ('in_production', 'progress'),
+         ('ready', 'planned'),
+         ], table='mrp_production', write='sql')
+    return draft_ids
+
+
+def create_moves_draft_mos(env, draft_ids):
+    env['mrp.production'].browse(draft_ids)._generate_moves()
+
+
+def map_mrp_workorder_state(cr):
+    legacy_state = openupgrade.get_legacy_name('state')
+    if openupgrade.column_exists(cr, 'mrp_workorder', legacy_state):
+        # if mrp_operations was installed
+        openupgrade.map_values(
+            cr,
+            legacy_state, 'state',
+            [('draft', 'pending'),
+             ('starworking', 'progress'),
+             ('pause', 'ready'),
+             ], table='mrp_workorder', write='sql')
+
+
+def update_stock_warehouse(env):
+    """Check for manufacturing warehouses and update them.
+    :returns: a map (dict) to populate MO's picking types.
+              keys: location_dest_id
+              values: corresponding manu_type_id
+    """
+    cr = env.cr
+    cr.execute(
+        """
+        SELECT location_dest_id
+        FROM mrp_production
+        GROUP BY location_dest_id;
+        """
+    )
+    ids = [r[0] for r in cr.fetchall()]
+    loc_ids = env['stock.location'].browse(ids)
+    wh_ids = []
+    loc_wh_map = dict()
+    for loc in loc_ids:
+        wh_id = loc.get_warehouse()
+        if wh_id:
+            wh_ids.append(wh_id.id)
+            loc_wh_map[loc.id] = wh_id.id
+    # update manufacture_to_ressupply
+    if wh_ids:
+        cr.execute(
+            """
+            UPDATE stock_warehouse
+            SET manufacture_to_resupply = TRUE
+            WHERE id in %s
+            """, (tuple(set(wh_ids)),)
+        )
+    # Create a default picking_type for manufacturing, for each warehouse above
+    loc_pt_map = dict()
+    wh_pt_map = dict()
+    for wh in env["stock.warehouse"].browse(list(set(wh_ids))):
+        wh._create_manufacturing_picking_type()
+        wh_pt_map[wh.id] = wh.manu_type_id.id
+    for loc in loc_wh_map.keys():
+        loc_pt_map[loc] = wh_pt_map[loc_wh_map[loc]]
+    return loc_pt_map
+
+
+def populate_production_picking_type_id(cr, loc_pt_map):
```

Comment:
Old procurement.rule records with `action == 'manufacture'` need updated as well to the newly created manufacturing picking type. Otherwise new manufacturing orders created by procurements (e.g. scheduler from orderpoint rules on manufactured products) are created from WH/IN/ sequence, not MO/ sequence, resulting in completely wrong MO names and all the other issues that might come from wrong picking type. In other words, if not doing that, this method only updates old MO's to have correct picking type, but new ones will still be wrong.

Reply:
What do you mean by updating rules and picking types? What kind of change?

Reply:
I had to do the following, otherwise automatically created MO-s take WH/IN/ sequence, not MO sequence:
`UPDATE procurement_rule SET picking_type_id=7 WHERE action='manufacture';` where 7 is the ID of stock_picking_type record with code='mrp_operation', which gets created during migration at some point (per earlier follow-up fixes that are now part of this PR already) -- manufacture procurement rule has to use the correct picking type, as-is it is wrong (unset) and the code ends up with an incoming picking type, taking sequence from WH/IN/